nimbella/storage/plugin_manager.py

```diff
@@ -0,0 +1,35 @@
+from .plugins.abstract_storage_plugin import AbstractStoragePlugin
+import sys
+import logging
+import importlib
+from importlib import resources
+
+# Plugin lookup table: Plugin Provider Id -> Plugin Class
+PLUGINS = {}
+
+# Return plugin class from provider identifier.
+# Loads all local plugins before searching for provider plugin.
+def find_plugin(id):
+    if not PLUGINS:
+        import_plugins()
+        register_plugins()
+
+    logging.debug(f'Looking for plugin from identifier: {id}')
+    return PLUGINS.get(id)
```

Comment:
Note that we have a legacy ... customers whose storage creds do not include a provider field.   This should default to  GCS.   Not sure whether the handling for that case belongs here, just noting it.